README.md

```diff
@@ -730,6 +731,36 @@ params do
 end
 ```
 
+#### Custom Types
+
+Aside from the default set of accepted types (`String`, `Integer`, `Boolean`, etc.),
+any class can be used as a type so long as it defines a class-level `from\_param` method.
+This method must return an instance of the same type, or raise an exception to indicate
+the value was invalid. E.g.,
+
+```ruby
+class Color
+  attr_reader :value
+  def initialize(color)
+    @value = color
+  end
+  def self.from_param(value)
+    raise "Invalid color" unless %w(blue red green).include?(value)
+    new(value)
+  end
+end
+
+# ...
+
+params do
+  required :color, type: Color, default: Color.new('blue')
+end
+get '/stuff' do
+  # params[:color] is already a Color.
+  params[:color].value
+end
+```
```

Comment:
Run this "code" through RuboCop, just to keep things consistent, change " to ', add a space between methods, etc.